pkg/minikube/cluster/cluster.go

```diff
@@ -424,7 +424,14 @@ func getServicePortFromServiceGetter(services serviceGetter, service string) (in
 	if err != nil {
 		return 0, fmt.Errorf("Error getting %s service: %s", service, err)
 	}
-	return int(svc.Spec.Ports[0].NodePort), nil
+	nodePort := 0
+	if len(svc.Spec.Ports) > 0 {
+		nodePort = int(svc.Spec.Ports[0].NodePort)
+	}
+	if nodePort == 0 {
+		return 0, fmt.Errorf("Service %s does not have a node port. To have one assigned automatically, the service type must be NodePort or LoadBalancer, but this service is of type %s.", service, svc.Spec.Type)
```

Comment:
Yes it does. Documented at https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L2007-L2020 & also confirmed by testing locally.